subprojects/core/src/main/java/org/gradle/api/internal/tasks/options/OptionReader.java

```diff
@@ -95,16 +102,28 @@ private void loadClassDescriptorInCache(Object target) {
         return optionValues.value();
     }
 
-    private Collection<OptionElement> getOptionElements(Object target) {
-        List<OptionElement> allOptionElements = new ArrayList<OptionElement>();
+    private static final class OptionElementAndSignature {
+        final OptionElement element;
+        final MethodSignature signature;
+
+        OptionElementAndSignature(OptionElement element, MethodSignature signature) {
```

Comment:
Mark `signature` arg and field as @Nullable.  (There is no package-info.java for this package.)